aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -641,18 +645,34 @@ where
                 .get(RETRY_AFTER)
                 .map(parse_retry_after)
                 .transpose()?;
-            let resp = AggregationJobResp::get_decoded(http_response.body())
-                .map_err(Error::MessageDecode)?;
+            let length = http_response
+                .headers()
+                .get(CONTENT_LENGTH)
+                .map(parse_content_length)
+                .transpose()
+                .map_err(|e| Error::BadRequest(e.into()))?;
```

Comment:
Is `Error::BadRequest` appropriate here, since this would in fact be an invalid response from the helper? I'm not certain what errors are available in this case, but if we don't have `BadResponse` already then something generic is probably fine.

Reply:
Well, the main options are:
- Error::Internal
- Error::HttpClient
- Error::Http

I suppose Internal isn't bad, in that it's probably what the resulting status code will be.